pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -1140,26 +1152,69 @@ private void overrideOwnership(String serviceUnit, ServiceUnitStateData orphanDa
         }
     }
 
+    protected void waitForCleanups(boolean excludeSystemTopics, int timeoutInMillis) {
+        long started = System.currentTimeMillis();
+        while (System.currentTimeMillis() - started < timeoutInMillis) {
+            boolean cleaned = true;
+            for (var etr : tableview.entrySet()) {
+                var serviceUnit = etr.getKey();
+                var data = etr.getValue();
+
+                if (excludeSystemTopics && serviceUnit.startsWith(SYSTEM_NAMESPACE.toString())) {
+                    continue;
+                }
+
+                if (data.state() == Owned && data.dstBroker().equals(lookupServiceAddress)) {
+                    cleaned = false;
+                    break;
+                }
+            }
+            if (cleaned) {
+                try {
+                    MILLISECONDS.sleep(OWNERSHIP_CLEAN_UP_CONVERGENCE_DELAY_IN_MILLIS);
```

Comment:
Sorry, I'm not sure why need to sleep while all service units are all cleaned.

Reply:
I added a comment on line 1231 to explain this.

Reply:
Ok, thanks.